File: front/src/types/entities/Trailer.ts
import Department from "./Department";
import Organization from "./Organization";

export default interface Trailer {
  id: string;
  title: string;
  organization: Organization;
  department: Department;
}

export interface CreateTrailer {
  title: string;
  organizationId: string;
  departmentId: string;
}

export interface UpdateTrailer extends CreateTrailer {
  id: string;
}

File: back/Main.Core/Entities/Trailer.cs
using Shared.Core.Entities;

namespace Main.Core.Entities;

public class Trailer : Entity<Trailer>
{
    public required string Title { get; set; }

    public Organization? Organization { get; set; }
    public required Guid OrganizationId { get; set; }

    public Department? Department { get; set; }
    public Guid? DepartmentId { get; set; }

    public override void Update(Trailer entity)
    {
        Title = entity.Title;
        
        OrganizationId = entity.OrganizationId;
        DepartmentId = entity.DepartmentId;
    }
}